### PR TITLE
TC02 - "User can't login with blank 'Username' textbox"

### PR DESCRIPTION
**Pull request by viethuynh140499**

### CODE REVIEW

File: src/test/java/test_cases/TestCase02.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    generalPage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult);
+  }
+}
```
Comment (tanmle) [inline src/test/java/test_cases/TestCase02.java:27]: break line

File: src/test/java/test_cases/TestCase02.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import org.testng.annotations.Test;
+import page_objects.GeneralPage;
+import page_objects.LoginPage;
+
+public class TestCase02 extends BaseTest {
+  GeneralPage generalPage = new GeneralPage();
+  LoginPage loginPage = new LoginPage();
+
+  @Test(description = "User can't login with blank Username textbox")
+  public void test() {
```
Comment (tanmle) [inline src/test/java/test_cases/TestCase02.java:15]: what does `test()`?

File: src/test/java/test_cases/TestCase02.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package test_cases;
+
+import common.helpers.Constant;
+import model.User;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import page_objects.GeneralPage;
+import page_objects.LoginPage;
+
+public class TestCase02 extends BaseTest {
```
Comment (tanmle) [inline src/test/java/test_cases/TestCase02.java:10]: rename class to be more meaningful

**Review by tanmle — CHANGES_REQUESTED** [review]

**Comment by tanmle** [comment]
rename your PR title please, refer to An's titles

File: src/test/java/test_cases/LoginTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    String actualResult = homePage.getWelcomeMessage();
+    String expectedResult = "Welcome " + user.getUsername();
+
+    Assert.assertEquals(actualResult, expectedResult);
+  }
+
+  @Test(description = "User can't login with blank Username textbox")
+  public void TC02() {
+
```
Comment (tanmle) [inline src/test/java/test_cases/LoginTest.java:29]: why do we have an empty line here?

File: src/test/java/test_cases/TestCase02.java (hunk 1 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/TestCase02.java]: Yes, I broke the line

File: src/test/java/test_cases/TestCase02.java (hunk 2 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/TestCase02.java]: I removed class TestCase2 and I merge 2 class into LoginTests

File: src/test/java/test_cases/TestCase02.java (hunk 3 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/TestCase02.java]: Yes, I renamed it into LoginTests

File: src/test/java/test_cases/LoginTest.java (hunk 4 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTest.java]: Oh thank you, I deleted the blank line

**Comment by tanmle** [comment]
hmmmm, alert helper?

**Comment by viethuynh140499** [comment]
> hmmmm, alert helper?

Sorry , I move AlertHelp into BrowserHelper class

File: src/test/java/test_cases/LoginTests.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  public void TC01() {
+    User user = new User(Constant.USERNAME, Constant.PASSWORD);
+
+    homePage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = homePage.getWelcomeMessage();
+    String expectedResult = "Welcome " + user.getUsername();
+
+    Assert.assertEquals(actualResult, expectedResult);
```
Comment (fey031) [inline src/test/java/test_cases/LoginTests.java:24]: Add assert message

File: src/test/java/test_cases/LoginTests.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  public void TC02() {
+    User user = new User("", Constant.PASSWORD);
+
+    homePage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult);
```
Comment (fey031) [inline src/test/java/test_cases/LoginTests.java:37]: Add assert message

**Review by fey031 — CHANGES_REQUESTED** [review]

File: src/test/java/test_cases/LoginTests.java (hunk 5 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: Yes, I added assert message

File: src/test/java/test_cases/LoginTests.java (hunk 6 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: Yes, I added assert message

**Review by tanmle — APPROVED** [review]

**Review by tanmle — APPROVED** [review]

**Review by fey031 — APPROVED** [review]
LGTM

**Pull request merged**